### PR TITLE
Create a secure option for asset roots

### PR DESCRIPTION
**Pull request by tst-cjeffries**
Adds a `--secure` flag to `geconfigureassetroot` and `geupgradeassetroot` to remove world-writable permissions.

To test:

- Build and install.
- Verify that the fusion executable is owned by `root:gegroup` and has the following permissions: `-r-xr-sr-x`.
- Verify that both `/gevol/assets/.config` and `/gevol/assets/.userdata` both are world writable.
- Open fusion and attempt to add a provider or sever association.
- Verify that the operation succeeds.
- Repair the asset root with the secure flag: `geconfigureassetroot --repair --secure`
- Verify that both `/gevol/assets/.config` and `/gevol/assets/.userdata` both are not world writable.
- Open fusion and attempt to add a provider or server association.
- Verify that the operation succeeds.
- Run the tutorial data scripts (for good measure).

### CODE REVIEW

File: earth_enterprise/src/fusion/autoingest/geAssetRoot.h [hunk 1]
```diff
@@ -38,6 +38,7 @@ enum SpecialDir {
 };
 extern std::string Dirname(const std::string& assetroot, SpecialDir dir);
 extern int DirPerms(SpecialDir dir);
+extern int SecureDirPerms(SpecialDir dir);
```
Comment (tst-jzmolek) [inline earth_enterprise/src/fusion/autoingest/geAssetRoot.h:41]: Add `Copyright 2020 The Open GEE Contributors` near the top

File: earth_enterprise/src/fusion/config/FixAssetRoot.h [hunk 2]
```diff
@@ -35,6 +35,7 @@ extern void PromptUserAndFixOwnership(const std::string &assetroot, bool nopromp
 // correct owner and permissions
 // returns true if it had to chown any of the dirs
 extern bool MakeSpecialDirs(const std::string &assetroot,
-                            const geUserId &fusion_user);
+                            const geUserId &fusion_user,
+                            bool secure);
```
Comment (tst-jzmolek) [inline earth_enterprise/src/fusion/config/FixAssetRoot.h:39]: Add `Copyright 2020 The Open GEE Contributors` near the top

File: earth_enterprise/src/installer/install_fusion.sh [hunk 3]
```diff
@@ -889,6 +889,10 @@ fix_postinstall_filepermissions()
     chmod -R 755 $BASEINSTALLDIR_OPT/qt
     chmod 755 $BASEINSTALLDIR_ETC/openldap
 
+    #sgid enabled
+    chown $ROOT_USERNAME:$GROUPNAME $BASEINSTALLDIR_OPT/bin/fusion
+    chmod g+s $BASEINSTALLDIR_OPT/bin/fusion
+
```
Comment (tst-jzmolek) [inline earth_enterprise/src/installer/install_fusion.sh:895]: Add `Copyright 2020 The Open GEE Contributors` near the top

**Comment by tst-jzmolek** [comment]
The code that creates the Fusion RPM should also be updated to set the ownership and permissions like the install.

**Comment by tst-cjeffries** [comment]
Good catch.

File: earth_enterprise/src/fusion/autoingest/geAssetRoot.cpp [hunk 4]
```diff
@@ -32,23 +32,39 @@ struct WantPerms {
...
+
+// These are the secure permissions
+const WantPerms secure_special_dirs[] = {
+    {"",           0755},
+    {".config",    0775},
+    {".state",     0755},
+    {".userdata",  0775},
+    {".privatedb", 0700}
+};
+
```
Comment (tst-basilhuffman) [inline earth_enterprise/src/fusion/autoingest/geAssetRoot.cpp:53]: very minor: might be better to use `std::array`s

Comment (tst-cjeffries) [reply earth_enterprise/src/fusion/autoingest/geAssetRoot.cpp:53]: The only reason I didn't change the data type is that it is supposed to match up to an enum defined in the header.

Comment (tst-cjeffries) [reply earth_enterprise/src/fusion/autoingest/geAssetRoot.cpp:53]: If you think it would be worth it I can certainly change it though.

**Comment by tst-cjeffries** [comment]
RPM changes have been added/tested @tst-jzmolek

**Pull request merged**